generate_endf71.py

```diff
@@ -99,122 +100,206 @@ def sort_key(path):
+            'compressed_files': ['WMP_Library_v1.1.tar.gz'],
+            'file_type': 'wmp',
+            'compressed_file_size': 12,
+            'uncompressed_file_size': 17
+        }
+    }
+}
+
+compressed_file_size, uncompressed_file_size = 0, 0
+for r in release:
+    for p in ('neutron', 'photon', 'neutron_wmp'):
+        compressed_file_size += release_details[release][p]['compressed_file_size']
+        uncompressed_file_size += release_details[release][p]['uncompressed_file_size']
+
+download_warning = """
+WARNING: This script will download up to {} MB of data. Extracting and
+processing the data may require as much as {} MB of additional free disk
+space. This script downloads ENDF/B-VII.1 incident neutron ACE data and
+incident photon ENDF data from NNDC and convert it to an HDF5 library
+for use with OpenMC.
+""".format(compressed_file_size, uncompressed_file_size)
+
+
+# ==============================================================================
+# DOWNLOAD FILES FROM NNDC SITE
+
+if args.download:
+    print(download_warning)
+    for particle in args.particles:
+        details = release_details[release][particle]
+        for i, f in enumerate(details['compressed_files']):
+            url = details['base_url'] + f
+            if 'checksums' in details.keys():
+                checksum = details['checksums'][i]
+                downloaded_file = download(url,
+                                           output_path=download_path / particle,
+                                           checksum=checksum)
+            else:
+                downloaded_file = download(url,
+                                           output_path=download_path / particle,
+                                           )
+
+# ==============================================================================
+# EXTRACT FILES FROM TGZ
+
+if args.extract:
+    for particle in args.particles:
+
+        if release_details[release][particle]['file_type'] == 'wmp':
+            extraction_dir = wmp_files_dir
+        elif release_details[release][particle]['file_type'] == 'endf':
+            extraction_dir = endf_files_dir
+
+        for f in release_details[release][particle]['compressed_files']:
+
+            # Extract files different depending on compression method
+            if f.endswith('.zip'):
+                with zipfile.ZipFile(download_path / particle / f, 'r') as zipf:
+                    print(f'Extracting {f}...')
+                    zipf.extractall(extraction_dir)
+            else:
+                with tarfile.open(download_path / particle / f, 'r') as tgz:
+                    print(f'Extracting {f}...')
+                    # extract files ignoring the internal folder structure
+                    for member in tgz.getmembers():
+                        if member.isreg(): 
+                            member.name = Path(member.name).name
+                            tgz.extract(member,path=extraction_dir)
```

Comment:
This doesn't really look necessary since we recursively glob for the files anyway

Reply:
The think was that this allows extraction to a particular dir name of our choosing. The internal folder structure of the WMP puts everything in a WMP_library folder. But yes the use of rglob would still find the files so I can remove this complexity